lib/src/data/data.rs

```diff
@@ -8,6 +8,7 @@ use std::time::Duration;
 use super::data_stream::{DataStream, kill_stream};
 use super::net_stream::NetStream;
 use ext::ReadExt;
+#[cfg(unix)] use unix_socket::UnixSocketStream;
```

Comment:
Putting `cfg(unix)` everywhere seems a bit unfortunate. Can we maybe put all of this functionality (or most of it) in a single module and expose exactly what we need?